content/blog/multi-cluster-consistency-models.md

```diff
@@ -0,0 +1,381 @@
+
+### Reading from the virtual stream
+Except for streams where the 'direct get' option is enabled (e.g. KV buckets) where direct get operations are automatically directed to any of the nodes within the local cluster replicating a mirror of the stream, if a client application wants to interact with a locally mirrored or sourced stream it needs to know the name of local stream, which means that it needs to know which region it is connected to. Avoiding this constraint is just like transparently dealing with publications to the virtual stream and can be done by setting a few (cluster-scoped) subject mapping transformations for the account at the Core NATS level.
+
+Besides the aforementioned direct get requests the way client applications 'read' (or consume) messages from a stream is through creating JetStream consumers (shared or not) and that is implemented over a number of JetStream API subjects which (unless JS domains are used) start with `$JS.API`, and also contain either a stream name or a consumer name as a token of that subject. Such that requests to create consumers on a stream `foo` are transparently transformed into requests to create consumers on the local `<region>-read-foo` stream instead.
+
+So for example: define a cluster-scoped subject mapping from `"$JS.API.CONSUMER.CREATE.foo.*"` to `"$JS.API.CONSUMER.CREATE.foo-read-west.{{wildcard(1)}}"` on cluster `west` such that any application connected to that cluster and creating a consumer on stream `foo` will create a consumer on stream `foo-read-west`.
+
+## What you can NOT do with a virtual stream
+- The retention policy of the virtual stream can not be a 'working queue' or 'interest' (i.e. only 'limits').
+- It does not work for KV buckets unless you know that you are not modifying the same key at the same time (or during a split brain) from two different regions.
+- Stream consumers are 'per region', meaning you do not have a global named durable consumer on the virtual stream, but multiple regional ones.
+- Deleting individual messages from a virtual stream is not possible, the delete operation will only apply to the local 'read' stream and are not propagated (and neither should they be, as the message sequence numbers are not homogeneous between regions).
+- Compare-and-set operations are not possible on the virtual stream as you would be reading from one stream and writing to another and the message sequence numbers are not preserved between them.
+
+# Walkthrough
+In this example, we're going to walk through setting up a local Super-Cluster and creating a virtual stream 'foo' 
+Make sure to install (or upgrade to) the latest version of the [NATS server](https://github.com/nats-io/nats-server/releases/latest) and of the [`nats` CLI tool](https://github.com/nats-io/natscli#installation) on your local machine.
+
+```bash
+git clone https://github.com/ConnectEverything/eventually_consistent_global_stream.git
```

Comment:
URLs are more natural with hyphens, maybe switch to that instead for the example repo?
```suggestion
git clone https://github.com/ConnectEverything/eventually-consistent-global-stream.git
```

Reply:
Done. Moved the files to a new public repo in the Synadia labs org where it belongs more